ksql-engine/src/test/java/io/confluent/ksql/util/AvroUtilTest.java

```diff
@@ -0,0 +1,172 @@
+        new TableElement("orderid", "BIGINT"),
+        new TableElement("itemid", "VARCHAR"),
+        new TableElement("orderunits", "DOUBLE"),
+        new TableElement("arraycol", "ARRAY<DOUBLE>"),
+        new TableElement("mapcol", "MAP<VARCHAR,DOUBLE>")
+        )));
+  }
+
+  @Test
+  public void shouldNotPassAvroCheckIfSchemaDoesNotExist() throws Exception {
+    SchemaRegistryClient schemaRegistryClient = mock(SchemaRegistryClient.class);
+    SchemaMetadata schemaMetadata = new SchemaMetadata(1, 1, null);
+    expect(schemaRegistryClient.getLatestSchemaMetadata(anyString())).andReturn(schemaMetadata);
+    replay(schemaRegistryClient);
+    AbstractStreamCreateStatement abstractStreamCreateStatement = getAbstractStreamCreateStatement
+        ("CREATE STREAM S1 WITH "
+         + "(kafka_topic='s1_topic', "
+         + "value_format='avro' );");
+    try {
+      avroUtil.checkAndSetAvroSchema(abstractStreamCreateStatement, new HashMap<>(), schemaRegistryClient);
+      fail();
+    } catch (Exception e) {
+      assertThat("Expected different message message.", e.getMessage(), equalTo(" Could not "
+                                                                               + "fetch the AVRO schema "
+                                                          + "from schema registry. null "));
+    }
+  }
+
+  @Test
+  public void shouldValidatePersistentQueryResultCorrectly()
+      throws IOException, RestClientException {
+    SchemaRegistryClient schemaRegistryClient = mock(SchemaRegistryClient.class);
+    KsqlTopic resultTopic = new KsqlTopic("testTopic", "testTopic", new KsqlAvroTopicSerDe
+        ());
+    Schema resultSchema = SerDeUtil.getSchemaFromAvro(ordersAveroSchemaStr);
+    PersistentQueryMetadata persistentQueryMetadata = new PersistentQueryMetadata("",
+                                                                                  null,
+                                                                                  null,
+                                                                                  "",
+                                                                                  null,
+                                                                                  DataSource.DataSourceType.KSTREAM,
+                                                                                  "",
+                                                                                  mock(KafkaTopicClient.class),
+                                                                                  new KsqlConfig(Collections.EMPTY_MAP),
+                                                                                  resultSchema,
+                                                                                  resultTopic,
+                                                                                  null);
+    expect(schemaRegistryClient.testCompatibility(anyString(), anyObject())).andReturn(true);
```

Comment:
Shouldn't the second argument be the schema we want to validate?

Reply:
Yes, made it more specific to get Schema class instance.